frontend/src/modules/WellCompletion/settings.tsx

```diff
@@ -0,0 +1,357 @@
+            }
+            if (Array.isArray(timeStepIndex)) {
+                let [startIndex, endIndex] = timeStepIndex;
+                if (startIndex > allTimeSteps.length) {
+                    startIndex = allTimeSteps.length - 1;
+                }
+                if (endIndex > allTimeSteps.length) {
+                    endIndex = allTimeSteps.length - 1;
+                }
+                timeStepIndex = [startIndex, endIndex];
+            }
+            if (timeStepIndex !== selectedTimeStepOptions.timeStepIndex) {
+                setSelectedTimeStepOptions((prev) => ({
+                    timeStepIndex,
+                    timeAggregation: prev.timeAggregation,
+                }));
+            }
+
+            if (!allTimeSteps || timeStepIndex === null || !selectedTimeStepOptions.timeAggregation) {
+                setPlotData(null);
+                return;
+            }
+            createAndSetPlotData(allTimeSteps, timeStepIndex, selectedTimeStepOptions.timeAggregation);
+        }
+    }, [wellCompletionQuery.data]);
+
+    React.useEffect(() => {
+        // Update plot when settings are adjusted, but no new query data is retrieved
+        if (
+            !wellCompletionsDataAccessor.current ||
+            !availableTimeSteps ||
+            selectedTimeStepOptions.timeStepIndex === null ||
+            !selectedTimeStepOptions.timeAggregation
+        ) {
+            return;
+        }
+
+        createAndSetPlotData(
+            availableTimeSteps,
+            selectedTimeStepOptions.timeStepIndex,
+            selectedTimeStepOptions.timeAggregation
+        );
+    }, [selectedTimeStepOptions, searchWellText, hideZeroCompletions]);
+
+    const createAndSetPlotData = React.useCallback(
+        function createAndSetPlotData(
+            availableTimeSteps: string[],
+            timeStepIndex: number | [number, number],
+            timeAggregation: TimeAggregation
+        ): void {
+            if (!wellCompletionsDataAccessor.current) {
+                setPlotData(null);
+                return;
+            }
+            if (typeof timeStepIndex === "number" && availableTimeSteps.length < timeStepIndex) {
+                setPlotData(null);
+                return;
+            }
+            if (
+                typeof timeStepIndex !== "number" &&
+                (availableTimeSteps.length < timeStepIndex[0] || availableTimeSteps.length < timeStepIndex[1])
+            ) {
+                setPlotData(null);
+                return;
+            }
+
+            const timeStepSelection: string | [string, string] =
+                typeof timeStepIndex === "number"
+                    ? availableTimeSteps[timeStepIndex]
+                    : [availableTimeSteps[timeStepIndex[0]], availableTimeSteps[timeStepIndex[1]]];
+            setPlotData(wellCompletionsDataAccessor.current.createPlotData(timeStepSelection, timeAggregation));
+        },
+        [wellCompletionsDataAccessor.current]
```

Comment:
Refs cannot be used in deps arrays (https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/). Consider having this without a callback and call it directly in the handler functions.